docs/docs/scheduled-triggers/recipes/check-expiration.mdx

```diff
@@ -296,8 +305,140 @@ nodemailer.createTestAccount((err, account) => {
 
 </details>
 
-You can run the server by running `node index.js` in your terminal. If you see the message
-`Webhook server is running on port 4000`, you're good to go!
+You can run the server by running `node index.js` in your terminal.
+
+</TabItem>
+<TabItem value="python">
+
+Make sure you have the necessary dependencies installed. You can use pip to install them:
+
+```bash
+pip install Flask secure-smtplib requests
+```
+
+<details>
+<summary>
+Then, create a new file called <code>index.py</code> and add the following code:
+</summary>
+
+```python
+from flask import Flask, request, jsonify
+import smtplib
+from smtplib import SMTPException
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+import requests
+from datetime import datetime, timedelta
+
+app = Flask(__name__)
+
+# Create an SMTP server connection
+def create_smtp_server():
+    try:
+        server = smtplib.SMTP(host='smtp.ethereal.email', port=587)
+        server.starttls()
+        server.login('example@superstore.com', 'samplePassword123')
+        return server
+    except SMTPException as e:
+        print(f"An error occurred while creating the SMTP server: {e}\nWe'll print the message to the terminal\n")
+        return None
+
+def get_expired_coupons():
+    today = datetime.now()
+    two_days_from_now = today + timedelta(days=2)
+    formatted_date = two_days_from_now.isoformat()
+
+    # Prepare the request headers
+    headers = {
+        'Content-Type': 'application/json',
+        'x-hasura-admin-secret': '<YOUR_ADMIN_SECRET>',
+    }
+
+    graphql_query = '''
+    query GetExpiringCoupons($two_days_from_now: timestamptz!) {
+      coupons(where: {expiration_date: {_lte: $two_days_from_now}}) {
+        id
+        expiration_date
+        code
+        user {
+          email
+          id
+          name
+        }
+      }
+    }
+    '''
+
+    graphql_variables = {
+        'two_days_from_now': formatted_date
+    }
+
+    # Prepare the request payload as a dictionary
+    payload = {
+        'query': graphql_query,
+        'variables': graphql_variables
+    }
+
+
+
+    # Make the POST request to the GraphQL endpoint
+    response = requests.post('<YOUR_PROJECT_ENDPOINT>', json=payload, headers=headers)
+
+    # Check the response status
+    if response.status_code == 200:
+        res = response.json()
+        return res['data']
```

Comment:
```suggestion
        return res['data']['coupons']
```

So, if we want the array (list) here, we need to target `coupons` instead of just `data`. This turns `expired_coupons` into a list that we _can_ loop over and target each dict (`coupon`) within the list so that we can destructure the properties and use them in the messages.

Otherwise, each `coupon` will just return `type <str>` in your loop.

Reply:
right